packages/react-server/src/ReactFizzServer.js

```diff
@@ -3658,11 +3770,12 @@ function flushSegment(
   request: Request,
   destination: Destination,
   segment: Segment,
+  rootBoundary: null | SuspenseBoundary,
```

Comment:
This feels leaky given that these can flush independently of any boundary. At least we should probably name this something much more esoteric and specific because this really means that the whole flushing mechanism is coupled to a very specific flushing strategy - which I'm not sure we will remember when we build SuspenseList etc. which will break some assumptions.